cloud/linode/client.go

```diff
@@ -27,3 +29,33 @@ type Client interface {
 
 // linodego.Client implements Client
 var _ Client = (*linodego.Client)(nil)
+
+func newLinodeClient(token, ua, apiURL string) (*linodego.Client, error) {
+	linodeClient := linodego.NewClient(nil)
+	linodeClient.SetUserAgent(ua)
+	linodeClient.SetToken(token)
+
+	// Validate apiURL
+	parsedURL, err := url.Parse(apiURL)
+	if err != nil {
+		return nil, err
+	}
+
+	validatedURL := &url.URL{
+		Host:   parsedURL.Host,
+		Scheme: parsedURL.Scheme,
+	}
+
+	linodeClient.SetBaseURL(validatedURL.String())
+
+	version := ""
+	matches := regexp.MustCompile(`/v\d+`).FindAllString(parsedURL.Path, -1)
```

Comment:
is this still required after the linodego update?

Reply:
This is the same logic as in the [linode-cosi-driver](https://github.com/linode/linode-cosi-driver/blob/main/pkg/linodeclient/linodeclient.go#L43-L76) and [linode-blockstorage-csi-driver](https://github.com/linode/linode-blockstorage-csi-driver/blob/main/pkg/linode-client/linode-client.go#L34-L75), added there just so we have similar code paths in all 3 components.

Reply:
maybe a dumb question, but why dont we make a `linodeClient.ParseURL` which does this logic and sets the base url and version for you. this would move all this logic to the linodego and we could reuse it right?

Reply:
@lgarber-akamai @zliang-akamai for visibility on my dumb questions

Reply:
Hi @luthermonson, I think you are right. Let us discuss it and get back to you next week.

Reply:
Sorry for the late follow-up, I think this is a good idea ^